cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -1058,6 +1058,22 @@ ArrayKernelExec GenerateTypeAgnosticPrimitive(detail::GetTypeId get_id) {
   }
 }
 
+/// similar to GenerateTypeAgnosticPrimitive, but for variable types
+template <template <typename...> class Generator>
+ArrayKernelExec GenerateTypeAgnosticVarBinaryBase(detail::GetTypeId get_id) {
+  switch (get_id.id) {
+    case Type::BINARY:
+    case Type::STRING:
+      return Generator<BinaryType>::Exec;
```

Comment:
Why don't you use `StringType` for `Type::STRING`? Fear of code duplication?

Reply:
Yeah, following the convention I was being used in the fill_na for primitives/temporals.

Reply:
Do you think it's overkill?

Reply:
No, it's ok. However, please don't use triple-slashes above for regular comments.